### DISCUSSION TITLE
export default is ignore? Empty output file

### DISCUSSION THREAD

**Discussion by baptisteArno** [webpack/webpack#13902] [Q&A]
Sorry, it's probably a noob question but here is what I'm trying to build:

```ts
import { initContainer } from "./embedTypes/container";
import { initPopup } from "./embedTypes/popup";
import { initBubble } from "./embedTypes/chat";

export default {
  initContainer,
  initPopup,
  initBubble,
};
```

`webpack.config.js`:

```js
const path = require("path");

module.exports = {
  mode: "production",

  entry: "./src/index.ts",

  output: {
    path: path.resolve(__dirname, "dist"),
    filename: "main.js",
  },

  resolve: {
    extensions: [".ts", ".js"],
  },

  module: {
    rules: [
      {
        test: /\.tsx?/,
        use: "ts-loader",
        exclude: /node_modules/,
      },
    ],
  },
};
```

`tsconfig.json`:
```json
{
  "compilerOptions": {
    "module": "ES2015",
    "target": "ES6",
    "moduleResolution": "node",
    "strict": true
  },
  "include": ["src/**/*.ts", "tests/**/*.ts"]
}
```

It outputs an empy `main.js` file and I don't understand why. Is this expected?

**Comment by baptisteArno** [comment]
Ok I just found the issue, it is the module prop in tsconfig that is invalid. I switched it to commonjs

**Reply by sokra** [reply]
That doesn't sound like the right answer. It seems like all your code was optimized away, because it's doesn't have any side effect. If you want to expose your exports you may want to use the `output.library` option.

**Reply by baptisteArno** [reply]
Indeed, you are right, adding the `output.library` option did the trick. Thank you @sokra
